### ISSUE TITLE
can't serialize nested associations

### ISSUE THREAD

**Issue by aphavichitr** [hauru/sequelize-to-json#7]
does this lib work with nested associations? i'm working with 2 level deep nested associations. for example models Order -> User -> UserLocation

json { order: { user: {userLocation:{ // i want data from here to be included in the json }}}

**Comment by hauru** [comment]
Yes, it was made with nested associations in mind. I think you should include assoc fields explicitely (either by their names or via `@assoc`) as they're not added by default.

**Comment by aphavichitr** [comment]
Do you have to add the field names with the association in front? For example user.id or just id? If userLocation also has an id field called id, wouldn't there be a conflict?

**Comment by hauru** [comment]
Well, there's an example in the [readme](https://github.com/hauru/sequelize-to-json#usage). Nested serialization should work regardless of the depth, as long as the schemes involved are set up properly.

**Comment by aphavichitr** [comment]
I think my issue has to do with having nested associations that return Arrays (one to many relationship) inside the outer models. I noticed in the example it only shows how to deal with one (.serialize) or many (.serializeMany) but not both if it's contained together in the nested layers.

**Comment by hauru** [comment]
Can you provide a sample of code that doesn't work?

**Comment by davesag** [comment]
I'm trying to serialise objects created using https://github.com/overlookmotel/sequelize-hierarchy

It works if I manually specify in the serialiser how deep to nest the hierarchy but I was hoping that just being able to do something like

```
const hierarchicalScheme = {
  include: ['@all', 'children'],
  exclude: ['@pk', '@fk', '@auto', 'hierarchyLevel'],
  assoc: {
    children: {
      include: ['@all', 'children'],
      exclude: ['@pk', '@fk', '@auto', 'hierarchyLevel']
    }
  }
}
```
would be enough to recurse the serialisation all the way down.  Alas if doesn't, it only does the first and second tiers.

**Comment by hauru** [comment]
@davesag Seems like useful capability. I'll think about it.

**Comment by davesag** [comment]
maybe something like
```
const hierarchicalScheme = {
  include: ['@all', 'children'],
  exclude: ['@pk', '@fk', '@auto', 'hierarchyLevel'],
  assoc: {
    children: {
      include: ['@parent']
    }
  }
}
```
